src/ConditionsResolver/Parser/ConditionalParameterParser.cs

```diff
@@ -243,29 +244,79 @@ private void ClearWorkflowParser()
         private (string? Result, ParameterContext Context) ResolveExecutionTasks(string value)
         {
             var subValue = value.Trim().Substring(ExecutionsTask.Length, value.Length - ExecutionsTask.Length);
-            var subValues = subValue.Split('[', ']');
+            var subValues = subValue.Split('.');
             var id = subValues[1].Trim('\'');
 
-            var task = WorkflowInstance?.Tasks.First(t => t.TaskId == id);
+            var task = WorkflowInstance?.Tasks.FirstOrDefault(t => t.TaskId == id);
 
-            if (task is null || task is not null && !task.Metadata.Any())
+            if (task is null)
             {
                 return (Result: null, Context: ParameterContext.TaskExecutions);
             }
 
-            var metadataKey = subValues[2].Split('\'')[1];
+            var subValueKey = subValues[2];
+            string? keyValue = null;
 
-            if (task is not null && task.Metadata.ContainsKey(metadataKey))
+            if (subValues.Length > 3)
             {
-                var result = task.Metadata[metadataKey];
+                keyValue = subValues[3]?.Split('\'')[1];
+            }
+
+            var resultStr = null as string;
+            switch (subValueKey.ToLower())
+            {
+                case "task_id":
+                    resultStr = task.TaskId;
+                    break;
+                case "status":
+                    resultStr = task.Status.ToString();
+                    break;
+                case "execution_id":
+                    resultStr = task.ExecutionId;
+                    break;
+                case "output_dir":
+                    resultStr = task.OutputDirectory;
+                    break;
+                case "task_type":
+                    resultStr = task.TaskType;
+                    break;
+                case "previous_task_id":
+                    resultStr = task.PreviousTaskId;
+                    break;
+                case "error_msg":
+                    resultStr = task.Reason.ToString();
+                    break;
+                case "result":
+                    resultStr = GetValueFromDictionary(task.ResultMetadata, keyValue);
```

Comment:
is this needed wouldn't TryGetValue work same here?

Reply:
I have used TryGetValue in that method, am keeping the method as it keeps this clean and provides nullchecks